paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -105,11 +106,14 @@ abstract class AbstractFileStoreTable implements FileStoreTable {
     @Nullable protected transient Cache<Path, Snapshot> snapshotCache;
     @Nullable protected transient Cache<String, Statistics> statsCache;
 
+    protected final ExternalPathProvider externalPathProvider;
```

Comment:
`ExternalPathProvider` is just created from options, it should be created in `AbstractFileStore.pathFactory()`.